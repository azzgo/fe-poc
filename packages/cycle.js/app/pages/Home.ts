import { h1 } from '@cycle/dom';
import { Sources } from '@cycle/run';
import xs from 'xstream';

export function HomePage(sources: Sources) {
  return {
    DOM: xs.of('').mapTo(h1('HOME')),
  };
}
