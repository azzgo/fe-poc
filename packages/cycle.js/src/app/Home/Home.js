import xs from 'xstream'
import { html } from 'snabbdom-jsx'
import { AppBar } from '../../shared/components/AppBar/AppBar'
import classNames from 'classnames'
import styles from './Home.less'
import { NoteCreator } from './components/NoteCreator/NoteCreator'
import { NoteCard } from './components/NoteCard/NoteCard'
import { actionTypes } from './constants'
import sampleCombine from 'xstream/extra/sampleCombine'
import isolate from  '@cycle/isolate'

function intent(source) {
  const allNotes$ = source.HTTP.select(actionTypes.fetchNotes)
    .flatten()
    .map((res) => res.body)
  
  const newNote$ = source.HTTP.select(actionTypes.createNote)
    .flatten()
    .map((res) => res.body)
  
  const deleteNote$ = source.HTTP.select(actionTypes.deleteNote).flatten()

  return xs.merge(
    allNotes$.map((notes) => ({ type: actionTypes.fetchNotes, payload: notes })),
    newNote$.map((newNote) => ({ type: actionTypes.createNote, payload: newNote})),
    deleteNote$.map(() => ({ type: actionTypes.deleteNote }))
  ).startWith({type: undefined})
}

function modal(notesActions$) {
  let state = {
    notes: []
  }

  return notesActions$.map((action) => {
    switch(action.type) {
    case actionTypes.fetchNotes:
      state = {
        notes: action.payload
      }
      break
    case actionTypes.createNote:
      state = {
        notes: state.notes.concat([action.payload])
      }
      break
    }

    return state
  })
}


function view(DOMSink, state$) {

  const appBar = AppBar()
  const noteCreator = NoteCreator({DOM: DOMSink})

  const noteCards$ = state$
    .map(({notes}) => {
      return notes.map((note) => {
        return isolate(NoteCard, note.id)({ DOM: DOMSink, props: { note, className: 'col-xs-4' }})
      })
    })

  return {
    vdom$: noteCards$
      .map((noteCards) => xs.combine(...noteCards.map((noteCard) => noteCard.DOM)))
      .flatten()
      .compose(sampleCombine(appBar.DOM, noteCreator.DOM))
      .map(([NoteCardViews, AppBarView, NoteCreatorView]) => (
        <div>
          {AppBarView}
          <div className={classNames(styles.warpper, 'row center-xs')}>
            <div className={classNames(styles.creator, 'col-xs-6')}>
              {NoteCreatorView}
            </div>
            <div className="notes col-xs-8">
              <div className="row between-xs">
                {NoteCardViews}
              </div>
            </div>
          </div>
        </div>
      )),
    request$: xs.merge(
      noteCreator.HTTP,
      noteCards$.map((noteCards) => xs.merge(...noteCards.map((noteCard) => noteCard.HTTP))).flatten()
    )
  }
}

export function Home(source) {
  const fetchNotes$ = xs.merge(
    xs.of({
      url: 'http://127.0.0.1:3000/notes',
      category: actionTypes.fetchNotes,
    }),
    source.HTTP.select(actionTypes.deleteNote)
      .flatten()
      .mapTo({
        url: 'http://127.0.0.1:3000/notes',
        category: actionTypes.fetchNotes,
      })
  )

  const actions$ = intent(source)
  const state$ = modal(actions$)
  const { vdom$, request$ } = view(source.DOM, state$)

  return {
    DOM: vdom$,
    HTTP: xs.merge(request$, fetchNotes$),
  }
}