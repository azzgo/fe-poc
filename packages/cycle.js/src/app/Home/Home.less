.warpper {
  margin-top: 50px;
}

.notes {
  padding-top: 50px;
}

.creator {
  margin-bottom: 40px;
}
