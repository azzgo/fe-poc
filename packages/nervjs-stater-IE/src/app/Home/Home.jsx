import React, { PureComponent } from 'react'
import { connect } from 'react-redux'

import classNames from 'classnames'

import { INote, NoteCard } from 'src/app/Home/components/NoteCard/NoteCard'
import NoteCreator from 'src/app/Home/containers/NoteCreator/NoteCreator'
import { deleteNoteAction, fetchNotesAction } from 'src/redux/ducks/notes'
import styles from './Home.less'

export class HomePage extends PureComponent {
  componentWillMount () {
    this.props.fetchNotes()
  }

  render () {
    return (
      <div className={classNames(styles.warpper, 'row center-xs')}>
        <div className={classNames(styles.creator, 'col-xs-6')}>
          <NoteCreator />
        </div>
        <div className="notes col-xs-8">
          <div className="row between-xs">
            {this.renderNoteCard()}
          </div>
        </div>
      </div>
    )
  }

  renderNoteCard = () => {
    return this.props.notes.map((note) => (
      <NoteCard key={note.id} className="col-xs-4" note={note} onChecked={this.onCheckCard} />
    ))
  }

  onCheckCard = (note) => {
    this.props.deleteNote(note)
  }
}

const mapStateToProps = (state) => ({
  notes: state.notes,
})

const mapDispatchToProps = {
  deleteNote: deleteNoteAction,
  fetchNotes: fetchNotesAction,
}

export default connect(mapStateToProps, mapDispatchToProps)(HomePage)
