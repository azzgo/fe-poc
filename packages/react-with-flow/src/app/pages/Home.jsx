import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'

export class Home extends PureComponent {
  render() {
    return (
      <div>
        Home
        <Link to="/about">Go to About</Link>
      </div>
    )
  }
}