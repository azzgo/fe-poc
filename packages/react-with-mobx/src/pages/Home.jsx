import {observer} from 'mobx-react';
import React from 'react';
import {counter, IncreaseCount} from '../store';

const Home = (props) => {
  return <>
    <h4>Home</h4>
    <p>counter: {counter.count}</p>
    <button onClick={() => IncreaseCount()}>+</button>
  </>;
};

export default observer(Home);
