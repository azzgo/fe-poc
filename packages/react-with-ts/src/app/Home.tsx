import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { returntypeof } from 'react-redux-typescript'

import { demoAction } from 'src/actions/demoAction'

interface IOwnProps {
}

interface IState {
}

const mapStateToProps = (state: IStoreState) => ({
  demoCount: state.demo,
})

const mapStateToPropsType = returntypeof(mapStateToProps)

type IStateToPropsType = typeof mapStateToPropsType

const mapDispatchToProps = {
  demoAction,
}

type IDispatchToPropsType = typeof mapDispatchToProps

@(connect(mapStateToProps, mapDispatchToProps) as InferableConnectType<IStateToPropsType, IDispatchToPropsType, IOwnProps>)
export class HomePage extends PureComponent<IOwnProps & IStateToPropsType & IDispatchToPropsType, IState> {
  private bindReference = (element: HTMLElement) => {
    console.log('bindReference', element)
  }

  render() {
    const { demoCount, demoAction } = this.props

    return (
      <div className="home">
        <p ref={(element) => console.log('arrow function', element) }>Count: {demoCount}</p>
        <button ref={this.bindReference}><Link to="about">about page</Link></button>
        <button onClick={() => demoAction()}>add count</button>
      </div>
    )
  }
}
