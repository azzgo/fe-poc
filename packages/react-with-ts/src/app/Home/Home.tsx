import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { returntypeof } from 'react-redux-typescript'

import classNames from 'classnames'

import { INote, NoteCard } from 'src/app/Home/components/NoteCard/NoteCard'
import NoteCreator from 'src/app/Home/containers/NoteCreator/NoteCreator'
import styles from './Home.less'
import { IStoreState } from 'src/redux/store'
import { deleteNoteAction, fetchNotesAction } from 'src/redux/ducks/notes'

interface IOwnProps {}

interface IState {}

const mapStateToProps = (state: IStoreState) => ({
  notes: state.notes,
})

const mapStateToPropsType = returntypeof(mapStateToProps)

type IStateToPropsType = typeof mapStateToPropsType

const mapDispatchToProps = {
  deleteNote: deleteNoteAction,
  fetchNotes: fetchNotesAction,
}

type IDispatchToPropsType = typeof mapDispatchToProps

type IProps = IOwnProps & IStateToPropsType & IDispatchToPropsType

@(connect(mapStateToProps, mapDispatchToProps) as InferableConnectType<IProps>)
export class HomePage extends PureComponent<IProps, IState> {
  public componentWillMount () {
    this.props.fetchNotes()
  }

  public render () {
    return (
      <div className={classNames(styles.warpper, 'row center-xs')}>
        <div className={classNames(styles.creator, 'col-xs-6')}>
          <NoteCreator />
        </div>
        <div className="notes col-xs-8">
          <div className="row between-xs">
            {this.renderNoteCard()}
          </div>
        </div>
      </div>
    )
  }

  public renderNoteCard = () => {
    return this.props.notes.map((note) => (
      <NoteCard key={note.id} className="col-xs-4" note={note} onChecked={this.onCheckCard} />
    ))
  }

  public onCheckCard = (note: INote) => {
    this.props.deleteNote(note)
  }
}
