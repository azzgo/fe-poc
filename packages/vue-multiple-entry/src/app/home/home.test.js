import { shallowMount } from '@vue/test-utils'
import Home from './home.vue'
import axios from 'axios'
import flushPromises from 'flush-promises'

test('test', () => {
  expect(1).toBe(1)
})

test('test2', async () => {
  axios.get.mockReturnValue(
    Promise.resolve({
      data: [
        {
          title: '测试title',
          body: '测试body',
        },
      ],
    })
  )
  const comp = shallowMount(Home)
  await flushPromises()
  expect(comp.html()).toMatchSnapshot()
})
