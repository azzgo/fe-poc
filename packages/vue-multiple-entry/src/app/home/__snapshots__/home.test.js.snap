// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test2 1`] = `
<div>
  <h1>Home Page</h1>
  <div>
    <h4>Post Data</h4>
    <article>
      <h5>测试title</h5>
      <div class="content">
        测试body
      </div>
    </article>
  </div>
</div>
`;
